### PR TITLE
build: cleanup Makefile

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**
* remove unnecessary bin target
* remove bin directory during clean
* fix typo
* do not clean entire go build cache (see https://github.com/golang/go/issues/53725)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -112,11 +108,12 @@ lint: build staticcheck ## run all linters
 
 .PHONY: clean
 clean: ## clean temporary files and directories
-	go clean -i -cache -testcache
```
Comment (AlexanderYastrebov) [inline Makefile:115]: -cache or -testcache flags remove entire go build cache.
https://github.com/golang/go/issues/53725 added warning about it which we silenced in #2217

**Review by alexandreliberato — APPROVED** [review]

**Comment by MustafaSaber** [comment]
👍

**Comment by AlexanderYastrebov** [comment]
:+1:

File: Makefile [hunk 2]
```diff
@@ -22,25 +22,21 @@ help: ## Display this help
 lib: $(SOURCES) ## build skipper library
 	go build ./...
 
-.PHONY: bindir
-bindir:
-	mkdir -p bin
```
Comment (szuecs) [inline Makefile:27]: That's not so great because it breaks the build for new contributors

Comment (AlexanderYastrebov) [reply Makefile:27]: No, bin is created by go build automatically

**Pull request merged**